packages/synapse-react-client/src/components/OAuthClientManagement/CreateOAuthClient.integration.test.tsx

```diff
@@ -47,18 +56,68 @@ function renderComponent(props: CreateOAuthModalProps = defaultProps) {
 
 function setUp(props: CreateOAuthModalProps = defaultProps) {
   const user = userEvent.setup()
+
+  // useFakeTimers to ensure that isOAuthClientReverificationRequired is called
+  // at least once per test, since the call is wrapped in useDebouncedEffect
+  jest.useFakeTimers()
+
   const component = renderComponent(props)
   const inputs = {
     name: screen.getByRole('textbox', { name: 'Client Name' }),
     homePage: screen.getByLabelText('Client Homepage'),
     redirectURI: screen.getByRole('textbox', { name: 'Redirect URI(s)' }),
+    sectorUri: screen.getByRole('textbox', { name: 'Sector Identifier URI' }),
+    userInfoSignedResponseAlg: screen.getByRole('combobox', {
+      name: 'User Info Signed Response Algorithm',
+    }),
   }
   const saveButton = screen.getByRole('button', { name: 'Save' })
+
+  act(() => jest.runOnlyPendingTimers())
+
+  // User real timers for the rest of the test
+  jest.useRealTimers()
+
   return { component, user, inputs, saveButton }
 }
 
+function mockVerificationPrecheckService(reverificationRequired: boolean) {
+  return rest.put(
+    `${getEndpoint(
+      BackendDestinationEnum.REPO_ENDPOINT,
+    )}/auth/v1/oauth2/client/${mockClient.client_id!}/verificationPrecheck`,
+    async (req, res, ctx) => {
+      return res(
+        ctx.status(200),
+        ctx.json({ reverificationRequired: reverificationRequired }),
+      )
+    },
+  )
+}
+
 describe('Create OAuth Client', () => {
   beforeAll(() => server.listen())
+  beforeEach(() => {
+    server.use(
+      rest.post(
+        `${getEndpoint(
+          BackendDestinationEnum.REPO_ENDPOINT,
+        )}/auth/v1/oauth2/client`,
+        async (req, res, ctx) => {
+          return res(ctx.status(200), ctx.json(req.body))
+        },
+      ),
+      rest.put(
+        `${getEndpoint(
+          BackendDestinationEnum.REPO_ENDPOINT,
+        )}/auth/v1/oauth2/client/${mockClient.client_id!}`,
+        async (req, res, ctx) => {
+          return res(ctx.status(200), ctx.json(req.body))
+        },
+      ),
+      mockVerificationPrecheckService(false),
```

Comment:
Most tests expect that `isOAuthClientReverificationRequired` will return `false`, so set that as the default value.